File: wechat/src/api/wechat/image.ts
import { httpPost } from '@/http/http'

// 微信图片上传响应类型
export interface WxImageUploadResponse {
  code: number
  data: {
    file: {
      id: number
      createdAt: string
      updatedAt: string
      deletedAt: string | null
      name: string
      classId: number
      url: string
      tag: string
      key: string
    }
  }
  msg: string
}

/**
 * 微信小程序上传图片
 * @param file - 要上传的文件
 * @param classId - 分类ID（可选）
 * @returns 上传结果
 */
export function wxUploadImage(file: File, classId?: number) {
  const formData = new FormData()
  formData.append('file', file)
  if (classId !== undefined) {
    formData.append('classId', classId.toString())
  }

  return httpPost<WxImageUploadResponse>('/api/wx/uploadImage', formData, undefined, {
    'Content-Type': 'multipart/form-data',
  })
}

File: gin-vue-admin/server/api/v1/wechat/image.go
package wechat

import (
	"strconv"

	"github.com/flipped-aurora/gin-vue-admin/server/global"
	"github.com/flipped-aurora/gin-vue-admin/server/model/common/response"
	exampleRes "github.com/flipped-aurora/gin-vue-admin/server/model/example/response"
	"github.com/flipped-aurora/gin-vue-admin/server/service"
	"github.com/gin-gonic/gin"
	"go.uber.org/zap"
)

type WechatImageApi struct{}

var fileUploadAndDownloadService = service.ServiceGroupApp.ExampleServiceGroup.FileUploadAndDownloadService

// WxUploadImage
// @Tags      WechatImage
// @Summary   微信小程序上传图片
// @accept    multipart/form-data
// @Produce   application/json
// @Param     file  formData  file                                                           true  "上传图片"
// @Param     classId  formData  int                                                           false  "分类ID"
// @Success   200   {object}  response.Response{data=exampleRes.ExaFileResponse,msg=string}  "上传图片,返回包括图片详情"
// @Router    /api/wx/uploadImage [post]
func (w *WechatImageApi) WxUploadImage(c *gin.Context) {
	_, header, err := c.Request.FormFile("file")
	if err != nil {
		global.GVA_LOG.Error("接收文件失败!", zap.Error(err))
		response.FailWithMessage("接收文件失败", c)
		return
	}

	// 获取并转换classId为整数
	classIdStr := c.DefaultPostForm("classId", "0")
	classIdInt, err := strconv.Atoi(classIdStr)
	if err != nil {
		classIdInt = 0 // 如果转换失败，默认为0
	}

	// 使用系统已有的文件上传服务
	file, err := fileUploadAndDownloadService.UploadFile(header, "0", classIdInt)
	if err != nil {
		global.GVA_LOG.Error("上传文件失败!", zap.Error(err))
		response.FailWithMessage("上传文件失败", c)
		return
	}
	response.OkWithDetailed(exampleRes.ExaFileResponse{File: file}, "上传成功", c)
}
